pkg/queue/queue.go

```diff
@@ -79,15 +79,15 @@ func NewRequestQueue(maxOutstandingPerTenant int, forgetDelay time.Duration, met
 // between calls.
 //
 // If request is successfully enqueued, successFn is called with the lock held, before any querier can receive the request.
-func (q *RequestQueue) Enqueue(tenant string, path []string, req Request, maxQueriers int, successFn func()) error {
+func (q *RequestQueue) Enqueue(tenant string, path []string, req Request, maxQueriers int, maxQueryCapacity float64, successFn func()) error {
```

Comment:
I find this to be a bit of a code smell. We shouldn't need to push down settings like this for each tenant through the call graph because we know the tenant ID and can retrieve it using `Overrides`, no?

Besides that, it just _feels_ strange to specify the capacity (or multiple capacity settings) when enqueueing a request; the original design may have been a bit flawed, but this seems like a good opportunity to correct it?

Reply:
totally agree with this. I update the pr with the suggested changes.